File: src/main/java/me/miki/shindo/management/profile/mainmenu/impl/ShaderBackground.java
package me.miki.shindo.management.profile.mainmenu.impl;

import me.miki.shindo.management.language.TranslateText;
import net.minecraft.util.ResourceLocation;

import java.io.File;

public class ShaderBackground extends Background {

    private final TranslateText nameTranslate;
    private final File shaderFile;
    private final ResourceLocation shaderResource;
    private int shaderId = -1;

    public ShaderBackground(int id, TranslateText nameTranslate, File shaderFile) {
        super(id, nameTranslate.getText());
        this.nameTranslate = nameTranslate;
        this.shaderFile = shaderFile;
        this.shaderResource = new ResourceLocation("shindo/shaders/menu.fsh");
    }

    public ShaderBackground(int id, TranslateText nameTranslate, ResourceLocation shaderResource) {
        super(id, nameTranslate.getText());
        this.nameTranslate = nameTranslate;
        this.shaderResource = shaderResource;
        this.shaderFile = null;
    }

    @Override
    public String getName() {
        return nameTranslate.getText();
    }

    public String getNameKey() {
        return nameTranslate.getKey();
    }

    public File getShaderFile() {
        return shaderFile;
    }

    public ResourceLocation getShaderResource() {
        return shaderResource;
    }

    public int getShaderId() {
        return shaderId;
    }

    public void setShaderId(int shaderId) {
        this.shaderId = shaderId;
    }

    public boolean isShaderLoaded() {
        return shaderId != -1;
    }

    public boolean hasResourceShader() {
        return shaderResource != null;
    }

    public boolean hasFileShader() {
        return shaderFile != null && shaderFile.exists();
    }
}

File: src/main/kotlin/me/miki/shindo/management/profile/mainmenu/impl/ShaderBackground.kt
package me.miki.shindo.management.profile.mainmenu.impl

import me.miki.shindo.management.language.TranslateText
import net.minecraft.util.ResourceLocation
import java.io.File

class ShaderBackground(
    id: Int,
    private val nameTranslate: TranslateText,
    private val shaderFile: File?,
    private val shaderResource: ResourceLocation,
) : Background(id, nameTranslate.getText()) {
    constructor(id: Int, nameTranslate: TranslateText, shaderFile: File) : this(
        id,
        nameTranslate,
        shaderFile,
        ResourceLocation("shindo/shaders/menu.fsh"),
    )

    constructor(id: Int, nameTranslate: TranslateText, shaderResource: ResourceLocation) : this(
        id,
        nameTranslate,
        null,
        shaderResource,
    )

    private var shaderId: Int = -1

    fun setShaderId(value: Int) {
        shaderId = value
    }

    override fun getName(): String = nameTranslate.getText()

    fun getNameKey(): String = nameTranslate.getKey()

    fun getShaderFile(): File? = shaderFile

    fun getShaderResource(): ResourceLocation = shaderResource

    fun getShaderId(): Int = shaderId

    fun isShaderLoaded(): Boolean = shaderId != -1

    fun hasFileShader(): Boolean = shaderFile != null && shaderFile.exists()
}
